api/src/wfl/module/copyfile.clj

```diff
@@ -37,8 +37,9 @@
 (defn create-copyfile-workload!
   "Use transaction TX to add the workload described by REQUEST."
   [tx {:keys [items common] :as request}]
-  (letfn [(merge-to-json [shared specific]
-            (json/write-str (util/deep-merge shared specific)))
+  (letfn [(nil-if-empty [x] (if (empty? x) nil x))
```

Comment:
I'm thinking if `(seq x)` will work here or not 🤔  ?

Reply:
perhaps not, `seq` will change the type of `x` so `json/write-str` might be affected.

Reply:
`seq` returns its input as a `seq` (whatever that is, some kinda sequence implementation) - we're trying to serialise to a json object not an array:

```clojure
(json/write-str (seq {:foo 1}))
=> "[[\"foo\",1]]"

(json/write-str {:foo 1})
=> "{\"foo\":1}"
```

Reply:
thanks for the explanation, yea I looked up what `deep-merge` returns and noticed it's not proper to use `seq` here

Reply:
first